### PR TITLE
Single TCP connection for TCK driver client

### PR DESCRIPTION
**Pull request by junaidkhalid**
Modified the javaClientDriver to open a single TCP connection, instead of opening a new TCP connection for every new request/stream.

### CODE REVIEW

File: rsocket-tck-drivers/src/main/java/io/rsocket/tckdrivers/client/JavaClientDriver.java [hunk 1]
```diff
@@ -519,7 +519,7 @@ private void handleCancel(String[] args) {
 
     private void handleEOF() {
         MySubscriber<Void> fnfsub = new MySubscriber<>(0L, AGENT);
-        RSocket fnfclient = createClient.get();
+        RSocket fnfclient = createClient;
```
Comment (yschimke) [inline rsocket-tck-drivers/src/main/java/io/rsocket/tckdrivers/client/JavaClientDriver.java]: nit: inline these

File: rsocket-tck-drivers/src/main/java/io/rsocket/tckdrivers/client/JavaClientDriver.java [hunk 2]
```diff
@@ -53,12 +53,12 @@
     private final Map<String, MySubscriber<Payload>> payloadSubscribers;
     private final Map<String, MySubscriber<Void>> fnfSubscribers;
     private final Map<String, String> idToType;
-    private final Supplier<RSocket> createClient;
+    private final RSocket createClient;
```
Comment (yschimke) [inline rsocket-tck-drivers/src/main/java/io/rsocket/tckdrivers/client/JavaClientDriver.java]: Can you call this just client.

**Comment by yschimke** [comment]
Failing on the TcpIntegrationTest I added conflicting ports, I'll fix that separately.

**Comment by yschimke** [comment]
I don't know how important it is to remove the Supplier, you could do this as a 1 line change by making your supplier just return a single instance.

**Comment by junaidkhalid** [comment]
I removed Supplier because I thought java.util.function.Supplier does not have memorization option. How can I make Supplier to return a single instance ?

**Comment by yschimke** [comment]
@junaidkhalid Supplier is a functional interface, so it can do anything.  I meant just

Client c = createClient();
Supplier<Client> sc = () -> c;

**Comment by yschimke** [comment]
@junaidkhalid I don't think I'm explaining myself well.  Sorry.

My point was because it is a supplier, you didn't need any changes in JavaClientDriver.java to support what you wanted.  You could make the supplier in JavaTCPClient.java return a constant value.

**Pull request merged**